src/components/swapv2/LimitOrder/ListOrder/useRequestCancelOrder.tsx

```diff
@@ -30,15 +31,16 @@ import { CancelOrderFunction, CancelOrderResponse, CancelOrderType, LimitOrder }
 const useGetEncodeLimitOrder = () => {
   const { account } = useActiveWeb3React()
   const [getEncodeData] = useGetEncodeDataMutation()
-  const { library } = useWeb3React()
+  const { readProvider } = useKyberSwapConfig()
```

Comment:
when using readProvider/library?

Reply:
- both of them is for interacting with rpc
- `readProvider` is for reading state of blockchain via rpc
- `library` is for signing transactions & send them to blockchain


- `readProvider` we use our custom rpc API
- `library` is readonly, return from web3-react/wallets e.g: metamask, rabby. We cannot customize the rpc API

For short:
- `readProvider` is for reading
- `library` is for writing

One more thing: from this PR, `library` will additional checking blackjack, `readProvider` not, letting user able to view the data. So we must use them correctly, if not, UI won't be able to show data correctly.